functions/java-jre.bash

```diff
@@ -17,60 +34,153 @@ java_webupd8_archive() {
+java_zulu_enterprise_8_apt(){
+  echo -n "$(timestamp) [openHABian] Installing Zulu Enterprise 64-Bit OpenJDK... "
+  cond_redirect apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 219BD9C9
+  if [ $? -ne 0 ]; then echo "FAILED (keyserver)"; exit 1; fi 
+  echo "deb http://repos.azulsystems.com/debian stable main" > /etc/apt/sources.list.d/zulu-enterprise.list
+  cond_redirect apt-get update
+  cond_redirect apt-get -y install zulu-8:
+  if [ $? -eq 0 ]; then echo "OK"; else echo "FAILED"; exit 1; fi  
+}
+
+## Fetches download link for Java Zulu 8.
+## Valid architectures: arm-64-bit, arm-32-bit-sf, arm-32-bit-hf, x86-32-bit, x86-64-bit
+##
+##    fetch_zulu_tar_url(String arch)
+##
+## TODO: Rewrite using download API when it leaves alpha state. https://www.azul.com/downloads/zulu-community/api/
+fetch_zulu_tar_url(){
+  local downloadlink
+  local filter
+  if [ ! -x "$(command -v jq)" ]; then
+    apt-get install -y jq
   fi
+  filter='.[] | select(.os == "Linux") | select(.category_slug == "java-8-lts") | select(.latest == 1) | select(.packaging_slug == "jdk") | select(.arch_slug == "'$1'") | select((.os_flavor | index("Debian")) or (.os_flavor == "[]")) | .["bundles"] | .[] | select(.extension == "tar.gz") | .["link"]' # $1 = e.g. "arm-64-bit"  
+  # Fetch an JSON array of download candidates from azul and filter them
+  # shellcheck disable=SC2006
+  downloadlink=`curl 'https://www.azul.com/wp-admin/admin-ajax.php' -s \
+        -H 'Accept: */*' \
+        -H 'X-Requested-With: XMLHttpRequest' \
+        -H 'Content-Type: multipart/form-data; boundary=---------------------------17447165291708986765346780730' \
+        -H 'Connection: keep-alive' \
+        -H 'Cache-Control: max-age=0' -H 'TE: Trailers' \
+        --data-binary $'-----------------------------17447165291708986765346780730\r\nContent-Disposition: form-data; name="action"\r\n\r\nbundles_filter_query\r\n-----------------------------17447165291708986765346780730\r\nContent-Disposition: form-data; name="action"\r\n\r\nsearch_bundles\r\n-----------------------------17447165291708986765346780730--\r\n' \
+        | jq -r "$filter"`
+  if [ -z "$downloadlink" ]; then return 1; fi
+    # shellcheck disable=SC2046,SC2005,SC2006
+    echo `echo "$downloadlink" | head -n1`
+  return 0   
 }
+
+## Check whatever a newer version Zulu version is available.
+## Returns 0 / true if new version exist
+java_zulu_tar_update_available(){
+  local availableVersion # format: "8u222-b10"
+  local javaVersion
+  local filter
+  if [ ! -x "$(command -v jq)" ]; then
+    apt-get install -y jq
+  fi
+  filter='[.[] | select(.os == "Linux") | select(.category_slug == "java-8-lts") | select(.latest == 1)  | .["openjdk_version"]] | first'
+  # Fetch an JSON array of download candidates from azul and filter them
+  # shellcheck disable=SC2006
+  availableVersion=`curl 'https://www.azul.com/wp-admin/admin-ajax.php' -s \
```

Comment:
this gives 222-b10 - which is correct for the non-ARM versions - but the ARM package installed returns the following:
<pre>
# /opt/jdk/zulu8.40.0.178-ca-jdk1.8.0_222-linux_aarch32hf/bin/java -version
openjdk version "1.8.0_222"
OpenJDK Runtime Environment (Zulu8.40.0.178-CA-linux_aarch32hf) (build 1.8.0_222-b178)
OpenJDK Client VM (Zulu8.40.0.178-CA-linux_aarch32hf) (build 25.222-b178, mixed mode, Evaluation)
<pre>

Reply:
Check out the end of the second row: `(build 1.8.0_222-b178)` there you have the match.
("222-b10" is just an example ofc.) But I will double check this function again.

Reply:
I tried. Azul provides different versions, 222-b10 ist current unless you select ARM....

Reply:
Hmm, I actually wonder if this issue can go "under the radar". Because this will only be applicable to x86 systems forcing the use of 32-bit Java. 64-bit systems will use the "apt-get" method.

And what will happen is just that Java will be reinstalled once more, nothing will break.
What do you think? It's nice to skip that extra logic / clutter in the code.